pkg/ruler/ruler.go

```diff
@@ -189,6 +191,8 @@ func (cfg *Config) RegisterFlags(f *flag.FlagSet, logger log.Logger) {
 
 	f.BoolVar(&cfg.EnableQueryStats, "ruler.query-stats-enabled", false, "Report the wall time for ruler queries to complete as a per-tenant metric and as an info level log message.")
 
+	f.Int64Var(&cfg.MaxGlobalRuleEvaluationConcurrency, "ruler.max-global-rule-evaluation-concurrency", 0, "Allow rules that don't have dependencies to be evaluated concurrently. 0 to disable.")
```

Comment:
I also would remove "global" from config. Why? Because in Mimir config "global" means "across the whole cluster" but here the real meaning is "per ruler process". I would just remove "global".